lib/data-store/message-handlers/helpers.js

```diff
@@ -15,8 +15,13 @@ var noopMessage = function noopMessage(dataStore, message) {
  * {@link https://api.slack.com/events/user_change|user_change}
  */
 var handleNewOrUpdatedUser = function handleNewOrUpdatedUser(dataStore, message) {
-  var user = new models.User(message.user);
-  dataStore.setUser(user);
+  var user = dataStore.getUserById(message.user.id);
+  if(user) {
+    user.update(message.user);
+  } else {
+    user = new models.User(message.user);
+    dataStore.setUser(user);
```

Comment:
Can you move the set outside the if block so it executes in either case? I'm going to be updating the way data stores function at some point so that a redis / memcache store could also be used.